File: algorithms/strings/camelcase/camelcase-javascript/js/testeCamelcase.js


import {camelcase} from '../js/camelcase.js';
import assert from 'assert';

describe('camelcase', function() {

    it('O número total de palavras encontradas na string é: 3', function() {

        let s = 'fraseComTres';
        let esperado = 3;
        
        let resultado = camelcase(s);
        
        assert.strictEqual(resultado, esperado);


    });
});

describe('camelcase', function() {

    it('O número total de palavras encontradas na string é: 4', function() {

        let s = 'fraseComQuatroPalavras';
        let esperado = 4;
        
        let resultado = camelcase(s);
        
        assert.strictEqual(resultado, esperado);


    });
});

describe('camelcase', function() {

    it('O número total de palavras encontradas na string é: 1', function() {

        let s = 'palavra';
        let esperado = 1;
        
        let resultado = camelcase(s);
        
        assert.strictEqual(resultado, esperado);


    });
});

describe('camelcase', function() {

    it('O número total de palavras encontradas na string é: 2', function() {

        let s = 'duasPalavras';
        let esperado = 2;
        
        let resultado = camelcase(s);
        
        assert.strictEqual(resultado, esperado);


    });
});

File: algorithms/strings/camelcase/camelcase-java/camelcase-java/src/testeCamelcase.java
import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class testeCamelcase {

    @Test
    public void testeCaso1() {

        String s = "fraseComTres";
        int esperado = 3;

        int resultado = Resultado.camelcase(s);

        assertEquals(resultado, esperado);
    }
    
    @Test
    public void testeCaso2() {

        String s = "fraseComQuatroPalavras";
        int esperado = 4;

        int resultado = Resultado.camelcase(s);

        assertEquals(resultado, esperado);
    }

    @Test
    public void testeCaso3() {

        String s = "palavra";
        int esperado = 1;

        int resultado = Resultado.camelcase(s);

        assertEquals(resultado, esperado);
    }
}
